### DISCUSSION TITLE
Views with Generic types

### DISCUSSION THREAD

**Discussion by holiday** [nalexn/ViewInspector#170] [Ideas]
Is there any future intention to add the ability to inspect generic view types? Views like these are used heavily in our code for reusability and we cannot inspect them at the moment. 

```Swift
public struct ExampleView<Content: View, Element: Equatable & Identifiable>: View {
    public var body: some View {
        ForEach(elements) { element in
            content(element)
        }
    }
}
```

**Comment by nalexn** [comment]
Hey! There should be no issues with inspecting views like that.

Traditionally there should be an `extension ExampleView: Inspectable { }`, and then you can either unwrap it explicitly via `inspect().view(ExampleView<MyView>.self)` or use `find(MyView.self)` to jump over the `ExampleView` inspection call all along. Note that `MyView` should also conform to the `Inspectable` protocol

**Reply by holiday** [reply]
Let me give that a shot! That worked, thanks a lot! 👍🏽

**Comment by taihuuho** [comment]
But a lot of time, you don't know, or it's hard to know what `Content` looks like

```swift
ExampleView {
    if a {
         Text("Hello World!")
           .background {...}
           ..
    }
    else {
         ...
   }
}
```

Then you would need to do `inspect().view(ExampleView<_ConditionalContent<ModifiedContent<ModifiedContent...>`
I don't think it's scalable enough.
